config/schemas/6/rapidast_schema.json

```diff
@@ -0,0 +1,605 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "title": "RapiDAST Configuration",
+    "description": "Schema for validating RapiDAST configuration files",
+    "type": "object",
+    "properties": {
+        "config": {
+            "type": "object",
+            "properties": {
+                "configVersion": {
+                    "default": 0,
+                    "$comment": "If no version is specified, the config converter will default to version 0",
+                    "type": "integer"
+                },
+                "base_results_dir": {
+                    "default": "./results",
+                    "type": "string"
+                },
+                "tls_verify_for_rapidast_downloads": {
+                    "default": true,
+                    "type": "boolean"
+                },
+                "environ": {
+                    "type": "object",
+                    "properties": {
+                        "envFile": {
+                            "type": "string"
+                        }
+                    },
+                    "required": []
+                },
+                "googleCloudStorage": {
+                    "type": "object",
+                    "properties": {
+                        "keyFile": {
+                            "type": "string"
+                        },
+                        "bucketName": {
+                            "default": "default-bucket-name",
+                            "type": "string"
+                        },
+                        "directory": {
+                            "type": "string"
+                        }
+                    },
+                    "comment": "Credentials are optional, as the client defaults to those inferred from the environment",
+                    "required": []
+                },
+                "defectDojo": {
+                    "type": "object",
+                    "properties": {
+                        "url": {
+                            "type": "string"
+                        },
+                        "authorization": {
+                            "type": "object",
+                            "properties": {
+                                "username": {
+                                    "type": "string"
+                                },
+                                "password": {
+                                    "type": "string"
+                                },
+                                "token": {
+                                    "type": "string"
+                                }
+                            }
+                        },
+                        "ssl": {
+                            "default": true,
+                            "type": "boolean"
+                        }
+                    },
+                    "oneOf": [
```

Comment:
Although the code [sets](https://github.com/RedHatProductSecurity/rapidast/blob/validate-config-jsonschema/rapidast.py#L353-L359) default empty values for the credentials if they are not provided, it seems reasonable to require either a username/password or a token

Reply:
+1